src/Nethermind/Nethermind.Crypto/KzgPolynomialCommitments.cs

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using Nethermind.Crypto.Properties;
+using Nethermind.Int256;
+
+namespace Nethermind.Crypto
+{
+    public static class KzgPolynomialCommitments
+    {
+        public static readonly UInt256 BlsModulus = UInt256.Parse("73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001", System.Globalization.NumberStyles.HexNumber);
+        public static readonly ulong FieldElementsPerBlob = 4096;
+
+        private const byte KzgBlobHashVersionV1 = 1;
+        private static IntPtr _ckzgSetup = IntPtr.Zero;
+
+        private static readonly ThreadLocal<SHA256> _sha256 = new();
+
+        private static readonly object _inititalizeLock = new();
+        public static void Inititalize()
+        {
+            lock (_inititalizeLock)
```

Comment:
Do you need that? I think runtime guarantees this?

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors

"The runtime calls a static constructor no more than once in a single application domain. That call is made in a locked region based on the specific type of the class. No additional locking mechanisms are needed in the body of a static constructor. To avoid the risk of deadlocks, don't block the current thread in static constructors and initializers. For example, don't wait on tasks, threads, wait handles or events, don't acquire locks, and don't execute blocking parallel operations such as parallel loops, Parallel.Invoke and Parallel LINQ queries."

Reply:
It's not a constructor, and it's not **in** constructor because we need control when it happens.

Reply:
It is called from `PointEvaluationPrecompile`, so why do we need to control this when we don't? Let it just be in static constructor?

Reply:
Let's add Initialization step for that